### ISSUE TITLE
Release django-sudo version 2.2

### ISSUE THREAD

**Issue by justinmayer** [mattrobenolt/django-sudo#20]
This issue is intended to track the packaged release of a new version of `django-sudo`. The proposed 2.2 version number is completely arbitrary and can of course be changed to whatever number is deemed most appropriate.

The primary rationale for creating this issue:

By subscribing to this issue, folks can be notified when the next release is available — i.e., when this issue is closed.

**Comment by mattrobenolt** [comment]
what why latest master is tagged at 2.1.0 already. There's nothing pending on master to be released.

**Comment by justinmayer** [comment]
Yes, I realize that. But when the time eventually comes, how will folks know when a new version of django-sudo is available? As far as I know, the only other way is for them to constantly come back every once in a while and check. This way, they can simply subscribe to this issue and be notified when it is closed. Does that help clarify?

**Comment by mattrobenolt** [comment]
I will literally never do this. There are many other ways to track new releases. GitHub itself provides an atom feed as one way for all releases: https://github.com/mattrobenolt/django-sudo/releases.atom

**Comment by justinmayer** [comment]
If you would be so kind as to humor a few additional thoughts…

The current packaged version of `django-sudo` (2.1.0) was released approximately 1.25 years ago and apparently does not function properly with Django 1.10 or 1.11, which incidentally are the only two versions that [Django currently supports](https://www.djangoproject.com/download/#supported-versions) in terms of bug fixes and security updates.

Given that the current packaged version of `django-sudo` does not appear to function properly on any supported version of Django, @moggers87 was kind enough to add support for Django 1.10 and 1.11 via pull request #18. That PR has, sadly, languished for the better part of a year. If there is anything blocking that PR’s merge and a subsequent packaged release, I'm not aware of any communication to that effect.

Ergo, please consider this a humble request for a speedy resolution to the aforementioned pull request, as well as a packaged release published to PyPI.

As an open source maintainer myself, I know all too well just how limited time can be to work on projects like this, so please don't take any of the aforementioned as opprobrium. On the contrary, I recognize how challenging it can be and am happy to help out here in any way I can.

To that end, is there anything I can do to help?

**Comment by mattrobenolt** [comment]
> Given that the current packaged version of django-sudo does not appear to function properly on any supported version of Django, @moggers87 was kind enough to add support for Django 1.10 and 1.11 via pull request #18. That PR has, sadly, languished for the better part of a year. If there is anything blocking that PR’s merge and a subsequent packaged release, I'm not aware of any communication to that effect.

I can spend the effort to review this again. I know it was lastly breaking support for Django 1.6 which is the only version I really care about personally. So it's a bit of a time sink to make sure we don't break support for what we need. I understand this is different needs from everyone else though.

**Comment by justinmayer** [comment]
> I know it was lastly breaking support for Django 1.6 which is the only version I really care about personally.

As far as I can tell, @moggers87 added back support for Django 1.6 on January 8th, one day after his initial pull request submission.

But that said, I completely understand that Django 1.6 support is your main focus, and ergo your reticence to do anything that may undermine it.

**Comment by mattrobenolt** [comment]
I maintain lots of open source projects, and I'm not going to check every single one of them if there's an issue open tracking a next release in each of them. That's silly when there are already tools to do this. It simply adds more maintenance burden. I've never seen a project do this, nor would I even have the audacity to propose them do this for my benefit as a consumer of their free software. This is literally what technology like Atom feeds are good for.

What happens if I want to tag 3.0 instead of 2.2? Lemme make sure to search the issue tracker on every project before I tag and make sure there's not an issue to close with it. I'll add this to my checklist of things to do for every project.

This is silly. And now super conflated with the open PR to add modern Django support. I am not keeping a ticket updated for tagging releases.

I'd be happy to look back at the other PR but it also drops other supported versions, which means this is not a major change and will need to be 3.0, not just 2.2.

**Comment by justinmayer** [comment]
I have seen projects do this, which is precisely where I got the idea to do it for the projects that I maintain. It is simply a minor step in the release process, whereby said issue is closed via the commit that issues the release, and a new issue is created at the same time the next-version-bump commit is pushed. I don't consider it a maintenance burden, particularly relative to the value it provides the community.

As I said in the original issue description, the proposed version number was completely arbitrary. Moreover, not including a number and just titling the issue as "Release next version" would address your concern about having to change the version number later.

Clearly, I don't think the proposed solution is "silly" since I'm the one who proposed it. But you've taken time out of your day to share your reasons why you don't care for the idea, which I appreciate.

**Comment by mattrobenolt** [comment]
I am locking this discussion now. :tada:
